test/caldav_client/tesla_test.exs

```diff
@@ -0,0 +1,32 @@
+defmodule CalDAVClient.TeslaTest do
+  use ExUnit.Case, async: true
+
+  describe "make_tesla_client/2" do
+    test "supports bearer auth" do
+      client = %CalDAVClient.Client{
+        server_url: "http://127.0.0.1:8800/cal.php",
+        auth: :bearer,
+        username: "",
+        password: "",
```

Comment:
It looks like in case of `:bearer` it is still necessary to pass `username: "", password: ""`.

Can you please refactor it so `username` and `password` fields are necessary only for `:basic` and `:digest` but for `:bearer` you only need to pass `token`?

I am wondering if this can be done so it's backward-compatible as well. If not, feel free to introduce new structs to represent Basic, Digest and Bearer authentication methods.